packages/drivers/odsp-driver/src/odspDeltaStorageService.ts

```diff
@@ -83,6 +85,7 @@ export class OdspDeltaStorageService {
                 headers: Object.keys(headers).length !== 0 ? true : undefined,
                 length: messages.length,
                 duration: response.duration, // this duration for single attempt!
+                SPResponseGuid,
```

Comment:
If exception is throw above (fetchAndParseAsJSON call), like when we timeout (abort fetch on timeout), then we never learn about this ID in our telemetry, and thus we do not have good way of asking service for logs.

I think two things should happen:
1. ID should be generated a bit deeper on the stack. For example, EpochTracker.fetchAndParseAsJSON adds epoch (by calling addEpochInRequest) - feels like perfect place to generate and add that ID there as well.
2. If exception is thrown, this ID should be added as a property to an error object thrown, such that it lands in telemetry eventually (by whoever logs it).

That would also has a nice benefit of more calls (including TreesLatest) benefitting from this flow.

As for logging on successful path - that's less interesting IMHO. Getting it here is fine, but we also have sprequestduration here, so it's less interesting / useful

Reply:
I think you refer to overwriting server generated correlation IDs. I do not think we should be doing that. Instead server-generated and client-generated IDs should exists in parallel, with both sub-systems recording in their telemetry pipelines both IDs. That way we can use it in production. If we are not using it in production, then there is very little usefulness, i.e. we should punt it.

Reply:
But this should have correlation with server id, right? This is the only way of specifying and in this case server replaces their guid with this guid. @marcmasmsft

Reply:
I will confirm once.

Reply:
I opened an issue here: https://github.com/microsoft/FluidFramework/issues/7175 to put an idea as to what Id we will put in the fetch request which the server can record in its telemetry and client can record the same. It will need server implementation too. So that entire work will take some time. For now, we have to use SPResponseGuid only and server will use the same id as SpRequestGuid. Currentlty doing it for only ops and that too on retry as it hamper a bit server ability to determine what farm/time the request was executed.